app/services/anime_search.rb

```diff
@@ -0,0 +1,59 @@
+class AnimeSearch
+  include ActiveModel::Model
+
+  SHOW_TYPES = %w(OVA ONA Movie TV Special Music)
+
+  SEASONS = %w(summer fall winter spring)
+
+  SEASONS_MONTHS = {
+    winter: %w(12 1 2),
+    spring: %w(3 4 5),
+    summer: %w(6 7 8),
+    fall:   %w(9 10 11)
+  }
+
+  AGE_RATINGS = %w{G PG13 PG R17+}
+
+  attr_accessor :name,
+                :id,
+                :show_type,
+                :age_rating,
+                :episode_count,
+                :started_airing,
+                :finished_airing,
+                :genres,
+                :season,
+                :year
+
+  with_options(allow_blank: true) do |item|
+    item.validates :show_type, inclusion: SHOW_TYPES
+    item.validates :season, inclusion: SEASONS
+    item.validates :age_rating, inclusion: AGE_RATINGS
+    item.validates :started_airing ,
+                   :finished_airing,
+                   format: {
+                     with: /\d{4}-\d{2}-\d{2}/,
+                     message: 'date must be formatted as YYYY-MM-DD'
+                   }
+    item.validates :year,
+                   format: {
+                     with: /\d{4}/,
+                     message: 'year must be formatted as YYYY'
+                   }
+  end
+
+  def apply_filters anime
+    anime = anime.full_search(name) unless name.nil?
+    anime = anime.where(slug: id) unless id.nil?
+    anime = anime.where(show_type: show_type) unless show_type.nil?
+    anime = anime.where(genres: { name: genres.split(',') }).joins(:genres) unless genres.nil?
```

Comment:
No, because I pass `Anime.preload(:genres)` to `apply_filters`. Also didn't put it inside the method so it will still preload with no filters